WordPress/gutenberg
# Add back button for isolated template part editor

## Description

Related to #33926. Addresses the following:

> - The editor canvas should isolate the Template Part (visually inset on the canvas to illustrate its boundaries).
> - Have a clear way to go back / out of the focused view.

Added a "edit-post"-style back button for isolated template part editor based on the design in https://github.com/WordPress/gutenberg/issues/29148#issuecomment-814941699, modified to include a dark background as seen in "edit-post" template editor. See the attached screencast below.

The back button is only visible when navigating to an "isolated template part editor" from another editor. It's not there in template editor, nor when landing directly in the isolated template part editor (as there isn't anywhere to go back to).

I noticed that we're not using browser's history to navigating between editors, I'm not sure if it's an oversight or an intentional behavior. Hence, we can't just use the native `history.back()` for this feature.

## How has this been tested?



1. Activate "tt1-block" theme.
2. Go to Site Editor.
3. Enable template part "focus mode" by selecting a template part and click "Edit %s" in the block toolbar.
4. Notice that there's dimmed background and a back button along with the previous template page name.
5. Click the back button to go back to the previous template.
6. Navigate to the template part editor via sidebar, notice the back button and the background are not there anymore.

## Screenshots 

https://user-images.githubusercontent.com/7753001/134610236-b7f01a34-fc9f-4c6d-bd20-996bfb2f6403.mp4

## Types of changes




New feature

## Checklist:
